src/notation/view/notationpaintview.cpp

```diff
@@ -388,20 +388,29 @@ std::pair<int, int> NotationPaintView::constraintCanvas(int dx, int dy) const
     QRectF viewport = this->viewport();
 
     QPoint canvasCenter = this->canvasCenter();
-    if (contentRect.width() <= viewport.width()) {
+
+    bool isScrollLimited = configuration()->isLimitCanvasScrollArea();
+
+    int offsetX = viewport.width() * 0.98;
+    int offsetY = viewport.height() * 0.98;
```

Comment:
Better to add some named constants for those values (static constexpr at the top of this file)

Reply:
Done. I should've done that first. Beginner mistake 😅